app/services/directions.js

```diff
@@ -13,9 +20,9 @@ export default function directions(from, to, mode) {
       travelMode: mode,
     }, (result, status) => {
       if(status === DirectionsStatus.OK) {
-        resolve(result.routes[0].legs[0].duration.value);
+        resolve([mode, result.routes[0].legs[0].duration.value]);
       } else {
-        reject(`Google returned error ${status}`);
+        resolve([mode, `Error: ${status}`]);
```

Comment:
This error handling is a bit sad, but I'm not sure what would be better. We could just `resolve(null);` in this case and `_.compact` those out, but that would mean the error reason getting lost. This way is also pretty skank - means we can't assume that the results are all successful, some could be successful and some errors like this.

Another options would be to have `directions` return an object whose values are all promises, but have a feeling that wouldn't be ideal either?